### PR TITLE
docs: detab generated docs

### PR DESCRIPTION
**Pull request by lewis6991**
- Replaced all tabs with spaces
    - vimdoc tags are not fully right justified
    - fixed formatting for option defaults than span multiple lines
    - refactored gen_eval_files.lua to handle whitespace more consistently

---

This also fixes the whitespace in the Lua `_meta` files since the rendered docs required `tabstop=8`, which shouldn't be required in code annotations.

### CODE REVIEW

**Comment by lewis6991** [comment]
Sorry reviewers.

**Comment by lewis6991** [comment]
@zeertzjq I've added a fixup that runs `retab!` on the help files. Is that an acceptable compromise?

**Comment by zeertzjq** [comment]
Not sure, but that seems to make reviewing easier.

**Comment by lewis6991** [comment]
Is reviewing your main concern?

**Comment by zeertzjq** [comment]
No, my main concern is that there should be no regressions, but noticing regressions requires reviewing to find out.

A problem I noticed here is that when some characters are concealed, using Tabs after them can keep the following content aligned, while using spaces cannot. And retabbing cannot change spaces in the middle of a line back to Tabs.

**Comment by lewis6991** [comment]
> A problem I noticed here is that when some characters are concealed

Most of these alignment regressions were in all the math functions (`cos`, `sin`, etc), but I fixed all that in the spaces version. Using `retab!` has re-introduced them again. However, these specific kinds of regressions aren't something I'm going to lose sleep over. IMO `<` shouldn't be mixed with text and should sit on its own line.

> using TABs after them can keep the following content aligned,

Not always. `gen_eval_files.lua` had some awkward logic to render tags with tabs. By using spaces and accounting for conceal characters manually we can align it exactly.

This is also true for the multiline option defaults which were fairly inconsistent and required fudging in `options.lua`. This PR removes most of that fudging and has a more consistent method for rendering them.

> And retabbing cannot change spaces in the middle of a line back to TABs.

I did say this was a compromise. There are maybe some regressions in places, but by rendering only with spaces allows us to be more precise about alignment and overall fixes more than it breaks. It's not like the original text was perfect before anyway.

**Comment by lewis6991** [comment]
The diff on the `.txt` files is now just:
```
 runtime/doc/builtin.txt | 1352 ++++++++++++++++++++++++++++++++++++++++++++++--------------------------------------------------
 runtime/doc/options.txt | 1849 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------------------------------
 runtime/doc/vvars.txt   |  336 ++++++++++++------------
 3 files changed, 1732 insertions(+), 1805 deletions(-)
```

**Comment by zeertzjq** [comment]
Hmm I made a mistake. `:retab!` did work for TAB alignment, however some spaces it has changed are not supposed to be TABs. I think it's possible to keep TABs in the middle of a line as TABs in Lua source files, and only change leading TABs (or TABs with only a `>` before), so it can be clear what spaces should be changed back to TABs.

File: scripts/gen_eval_files.lua [hunk 1]
```diff
@@ -337,9 +337,30 @@ local function render_api_keyset_meta(_f, fun, write)
...
+  local xs = split(x)
+  local leading_ws = xs[1]:match('^%s*') --[[@as string]]
+  local leading_ws_pat = '^' .. leading_ws
+
+  for i in ipairs(xs) do
+    local strip_pat = xs[i]:match(leading_ws_pat) and leading_ws_pat or '^%s*'
+    xs[i] = xs[i]:gsub(strip_pat, '')
+  end
+
+  return table.concat(xs, '\n')
```
Comment (zeertzjq) [inline scripts/gen_eval_files.lua:352]: This function is too complicated. Each file should just have a hard-coded indent to be stripped.

**Pull request closed without merging**